packages/apollo-collaboration-server/src/fileHandling/fileHandling.service.ts

```diff
@@ -468,4 +458,65 @@ export class FileHandlingService {
     )
     return resultJsonArray
   }
+
+  /**  Write cache into file
+   * @returns Filename where cache was written
+   */
+  async downloadCacheAsGFF3file(): Promise<string> {
+    let cacheValue: string | undefined = ''
+    try {
+      // Join path+filename
+      const { DOWNLOADED_OUTPUT_FOLDER } = process.env
+      if (!DOWNLOADED_OUTPUT_FOLDER) {
+        throw new Error('No DOWNLOADED_OUTPUT_FOLDER found in .env file')
+      }
+      const downloadFilename = join(
+        DOWNLOADED_OUTPUT_FOLDER,
+        `downloaded_${getCurrentDateTime()}.gff3`,
+      )
+      const nberOfEntries = await this.cacheManager.store.keys?.()
+      nberOfEntries.sort((n1, n2) => n1 - n2) // Sort the array
+      // Loop cache in sorted order
+      for (const keyInd of nberOfEntries) {
+        cacheValue = await this.cacheManager.get(keyInd.toString())
+        if (!cacheValue) {
+          throw new Error(`No entry found for ${keyInd.toString()}`)
+        }
+        this.logger.verbose(
+          `Write into file =${JSON.stringify(cacheValue)}, key=${keyInd}`,
+        )
+        // Write into file line by line
+        fs.appendFile(downloadFilename, gff.formatSync(JSON.parse(cacheValue)))
+      }
+      this.logger.debug(`Cache saved to file ${downloadFilename}' successfully`)
+      return downloadFilename
+    } catch (err) {
+      this.logger.error(`ERROR when saving cache to file: ${err}`)
+      throw new InternalServerErrorException(
+        `ERROR in downloadCacheAsGFF3file() : ${err}`,
+      )
+    }
+  }
+
+  /**
+   * Check if GFF3 is loaded into cache. Basically we check if number of entries > 0 then GFF3 is loaded. Otherwise not
+   * @returns TRUE: GFF3 is loaded into cache, otherwise return FALSE
+   */
+  async checkCacheKeys() {
+    try {
+      const nberOfEntries = await this.cacheManager.store.keys?.()
+      // Check if there is any item inside.
+      // Later we will check what kind of items there are in cache so we can check specifically if there is GFF3 data
+      for (const keyInd of nberOfEntries) {
+        return true
+      }
+      return false
+    } catch (err) {
+      this.logger.error(`ERROR in checkCacheKeys: ${err}`)
+      throw new HttpException(
```

Comment:
Should use `InternalServerErrorException` instead of `HttpException`

Reply:
Done